ONSdigital/ras-rm-survey
# Creating survey endpoints

# Motivation and Context
The survey endpoints needed to be implemented.

# What has changed
Implemented 5 survey endpoints:

* GET /survey
* POST /survey
* GET /survey/{reference}
* DELETE /survey/{reference}
* PATCH /survey/{reference}

Also created a unit test for each endpoint that checks the status code and the returned JSON.

# How to test?
* Run the service and try to hit each endpoint.
* Run `go test .` and make sure the tests pass (the health endpoint one has a habit of failing randomly; just keep retrying until it passes)

# Links
[Trello card](https://trello.com/c/Sj4DE0DT)
